plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -3084,10 +3084,10 @@ public void run(HiveMetastoreClosure metastore, AcidTransaction transaction)
                             // but another tx already created T(a: varchar)).
                             // This may be a problem if there is an insert after this step.
                             if (!hasTheSameSchema(newTable, table)) {
-                                e = new TrinoException(TRANSACTION_CONFLICT, format("Table already exists with a different schema: '%s'", newTable.getTableName()));
+                                failure = new TrinoException(TRANSACTION_CONFLICT, format("Table already exists with a different schema: '%s'", newTable.getTableName()));
                             }
-                            else {
-                                done = ignoreExisting;
+                            else if (!ignoreExisting) {
```

Comment:
I think this is inverted. In the previous code, we have `done=true` if `ignoreExisting==true`. So we would want to set `failure=null` in that case.